### PR TITLE
Add SATSCARD parser regression tests

### PR DESCRIPTION
**Pull request by diyaayay**
## Summary

Add focused SatsCard URL parse tests covering unsealed state, empty and unknown state, missing fields (nonce, signature, state, address, ident), invalid slot number, short signature, invalid host, and leading whitespace.

### Platform Coverage

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on iOS simulator
- [ ] Tested on Android simulator
- [ ] Not tested

## Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/bitcoinppl/cove/blob/master/CONTRIBUTING.md)
- [x] I have read [ARCHITECTURE.md](https://github.com/bitcoinppl/cove/blob/master/ARCHITECTURE.md)

## Summary by CodeRabbit

* **Bug Fixes**
  * Added stricter signature validation to reject malformed/too-short signatures before processing.

* **Tests**
  * Expanded test coverage for SATSCARD URL parsing, covering missing/empty/unknown fields, invalid numbers, host validation, and trimming whitespace.
  * Added tests for message construction, identifier conversion error mapping, and tap-signer parsing failure cases.

### CODE REVIEW

File: rust/crates/cove-tap-card/src/parse.rs [hunk 1]
```diff
@@ -321,4 +321,138 @@ mod tests {
...
+        let err = TapCard::parse(&satscard_https(&fragment)).unwrap_err();
+        assert!(matches!(err, Error::MissingField(Field::Signature)));
+    }
+
+    #[test]
+    fn test_sats_card_missing_state_errors() {
+        let fragment = "o=0&r=95kesdwq&n=ab78fd50637f8f5a&s=26d1a0684f99fe43b223dca75081bb05bd0233b901139cdd33a4d0a2e61666ed1470d7c53d90f6ae4c60a6cbc7a0f4ded5f13461092b24604ad476bbcf1dd913";
+        let err = TapCard::parse(&satscard_https(fragment)).unwrap_err();
+        assert!(matches!(err, Error::MissingField(Field::State)));
+    }
```
[the remark this answers is not in the document]
Comment (devvaansh) [reply rust/crates/cove-tap-card/src/parse.rs:358]: agree!, @diyaayay Use SATSCARD_SEALED_FRAGMENT.replacen("u=S&", "", 1) instead

[the remark this answers is not in the document]
Comment (diyaayay) [reply rust/crates/cove-tap-card/src/parse.rs:358]: > agree!, @diyaayay Use SATSCARD_SEALED_FRAGMENT.replacen("u=S&", "", 1) instead

Makes sense! I'll change it. Thanks:)

**Review by praveenperera — APPROVED** [review]
thanks @diyaayay approved!
